### ISSUE TITLE
3D measurements seems to be impaired 

### ISSUE THREAD

**Issue by dfgdgdfgd** [SchmollerLab/Cell_ACDC#252]
### Description:

Hi Francesco,

i had to install a fresh windows version and since that, i get during the start of ACDC always the message, that setuptools needs to be installed.

![image](https://user-images.githubusercontent.com/116281469/214280556-0eca0492-1e89-434a-9c70-371f81f1d316.png)

But the console shows the following errors:

![image](https://user-images.githubusercontent.com/116281469/214281119-07eb0fd4-054d-4383-9055-bb319e76b623.png)

Anyway, ACDC is working, but if i try to compute measurements, most of the 3D metrics are not available (which were before using the same datasets).....

![image](https://user-images.githubusercontent.com/116281469/214281643-b3148f53-6e5d-43a8-949c-d0fbceb34fea.png)

I guess, that is correlated to the setuptools issue, isn´t it?

Thanks & best regards
Mario

**Comment by ElpadoCan** [comment]
Hi Mario, thanks for reporting this, I have never seen that setuptools error message, but it should not affect Cell-ACDC at all. I will add proper handling of this error.  

However, I don't see how this is correlated to the missing 3D measurements. When you run the GUI and you get the pop-up message where you insert the metadata such as the number of z-slices, is the `3D segmentation (z-stacks)` toggle moved to the right (meaning that the segmentation file is actually 3D)? Send me a screenshot of that dialog please, thank you very much

**Comment by dfgdgdfgd** [comment]
Hi Francesco,

thanks for taking care so quick!
Yes, the toggle is moved to the right...


![image](https://user-images.githubusercontent.com/116281469/214293855-c344aced-c0da-4e72-a2dd-466f0ad505dc.png)

I now have this problem on another computer after updating ACDC to version .59.....
Is it possible to downgrade it to see if it´s a bug within the new version?

**Comment by dfgdgdfgd** [comment]
ok, that´s strange....not all 3D metrics are unavailable......?

![image](https://user-images.githubusercontent.com/116281469/214294874-c785701e-aa1a-45b1-b94f-a4370fbb18b2.png)

**Comment by ElpadoCan** [comment]
Oh wait, now I know what you mean. You are talking about the greyed-out metrics? 

Ok, that's a new feature because some users were disabling the background value while keeping the amount values for example. That would cause Cell-ACDC to actually not save the amount values because it needs the background value to do that. 

So now it is not possible to disable the background value median if you require amount metrics. You see that the background value is disabled BUT it is checked because you are requesting to save the amount (hence background cannot be unchecked). It doesn't mean it is not available, it simply means that you cannot change its status, i.e., you cannot choose to NOT save it.

For the data prep values instead, you cannot choose to save them since you did not perform background selection in the data prep step. This is of course totally fine, but since there are no background ROIs available then it was confusing that you could still choose to save it. In previous versions, you could choose to save data prep values (amount and concentration) even if you didn't select the background ROIs, but Cell-ACDC was not saving them (since it was missing the required info, i.e., the background ROIs).

Now that I think about it, it is still a bit confusing, however, this is explained in the info button related to the metric. It should tell you why it is greyed out.

To summarise:

- If you request metrics that require background value (median) like the amount, then the background value median will be checked (and saved), but disabled so that the user cannot uncheck it (causing the related metrics to be ignored).
- If you did not select background ROIs in the data prep step then you cannot choose to save data prep metrics

Let me know if it's clear and also if you have ideas on how to improve the UI to avoid this confusion.

PS: I will soon release a version with a fix for the setuptools issue, but as I said, it should not affect any functionality

**Comment by dfgdgdfgd** [comment]
oh, yeah, damn, you're right. I was so convinced it was an installation error that I didn't even do the measurements on a test run.... ;)  sorry for the red alert..... ;)

I didn´t realized, that you uploaded a new version with those changes in the meantime. 

For the users it would maybe be helpful to keep the overwiew on new features/improvements, if you add an update info on your homepage (i guess it is much more complex to build a version check into the software??), what do you think?

Anyway, thank´s again for solving the "problem" ;) so quick,
best
Mario

**Comment by ElpadoCan** [comment]
Yes, I totally agree, we will need proper release notes at some point. 

We are working on a website where we will write all the announcements, the "What's new" page, and the roadmap so that users know what to expect from the future. Plus all the instructions, how to get started, tutorials etc. so stay tuned for that :D 

Cheers,
Francesco

PS: for this specific issue I decided that instead of disabling metrics Cell-ACDC will show a pop-up message with the explanation why a specific metric cannot be checked or unchecked, what do you think?

**Comment by dfgdgdfgd** [comment]
Cool, sounds great, looking forward to it! :)

Sure, you can do it that way, but to be honest, if I had bothered to click on the question mark, it would have been clear to me why it doesn't work.... ;)
If you want to bother with the hints, it's definitely foolproof..... :D

**Comment by ElpadoCan** [comment]
Hi Mario, I released `v1.2.4rc60` with a fix for the `setuptools` issue. 

Since I had to fix the `setuptools` issue, I also added what we discussed. Cell-ACDC will now show a pop-up to explain why a metric cannot be checked or unchecked instead of disabling it. Hopefully, it will avoid confusion now.

Have a good day! 
Francesco

**Comment by dfgdgdfgd** [comment]
Hi Francesco,

ok, that was really fast 👍 
I´m sure, now it´s clear for all..... :)

Have also a good day,
ciao, Mario
